eth/gasestimator/gasestimator.go

```diff
@@ -82,12 +83,15 @@ func Estimate(ctx context.Context, call *core.Message, opts *Options, gasCap uin
 				if err != nil {
 					return 0, nil, err
 				}
+			} else {
+				available = alternativeFeeBalance
 			}
-		} else {
-			if call.Value != nil {
-				if call.Value.Cmp(available) >= 0 {
-					return 0, nil, core.ErrInsufficientFundsForTransfer
-				}
+		}
+		if call.Value != nil {
+			if call.Value.Cmp(celoBalance) >= 0 {
+				return 0, nil, core.ErrInsufficientFundsForTransfer
+			}
+			if call.FeeCurrency == nil || !call.IsFeeCurrencyDenominated() {
```

Comment:
This `||` is incorrect, the `IsFeeCurrencyDenominated` is mainly an indicator to separate CIP-64/CIP-66, and a CIP-66 will have `call.IsFeeCurrencyDenominated() == false` and `bool(call.FeeCurrency != nil) == true`.
With this you will subtract a native `call.Value` from an `available` fee-currency value.

Given that the notion of `IsFeeCurrencyDenominated` is also incorrect above, I think there was a misunderstanding of that getter.

Reply:
Ok, I've took it as Celo, but the only requirement is to translate the prices to the feeCurrency. Got it